modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -178,6 +184,31 @@ object ConsensusManager {
             .handleErrorWith(logger.error(_)(s"Error checking for consensus state update {key=${key.show}}"))
         }.void
 
+      private def filterPeerList(peers: List[Peer]): F[List[Peer]] = {
+        case class PeerSnapshot(ordinal: SnapshotOrdinal, hash: Hash, peer: Peer)
+        val groupFn: PeerSnapshot => (SnapshotOrdinal, Hash) = v => (v.ordinal, v.hash)
+
+        peers
+          .traverse(peer =>
+            gsClient.getLatest(peer).flatMap { signedSnapshot =>
+              signedSnapshot.toHashed.map(_.hash).map(PeerSnapshot(signedSnapshot.value.ordinal, _, peer))
+            }
+          )
+          .map(_.groupBy(groupFn))
+          .map(_.maxBy(_._2.size)._2.map(_.peer))
+      }
+
+      private def getPeerSublist(peers: List[Peer]) = {
+        val maxSublistPercent = 0.25
+        val maxSublistSize = (peers.size * maxSublistPercent).toInt + 1
```

Comment:
So you want at most `0.25` of the `peers.size` but at least `1` as you wrote in the comment:
https://github.com/Constellation-Labs/tessellation/pull/616#discussion_r1164234646

According to that statement for `peer.size = 8` it should return `2` but your logic returns `3` which is not `0.25` but `0.375`.

To make the logic working just like you explained you should rather 
```scala
val maxSublistSize = Math.max((peers.size * maxSublistPercent).toInt, 1)
```

Reply:
Changed.